domain/sysvar_cache.go

```diff
@@ -252,6 +252,14 @@ func (do *Domain) checkEnableServerGlobalVar(name, sVal string) {
 		variable.StatsLoadPseudoTimeout.Store(variable.TiDBOptOn(sVal))
 	case variable.TiDBTxnCommitBatchSize:
 		storekv.TxnCommitBatchSize.Store(uint64(variable.TidbOptInt64(sVal, int64(storekv.DefTxnCommitBatchSize))))
+	case variable.TiDBStatsCacheMemQuota:
+		var val int64
+		val, err = strconv.ParseInt(sVal, 10, 64)
+		if err != nil {
+			break
+		}
+		variable.StatsCacheMemQuota.Store(val)
+		do.SetStatsCacheCapacity(val)
```

Comment:
We just refactored this in https://github.com/pingcap/tidb/pull/33897

Unfortunately if you need the domain, you'll need to specially handle this. I don't think there's an easy way to access it from the SetGlobal func.

Reply:
updated.